reverse.go

```diff
@@ -110,21 +110,29 @@ func commandReverse(args []string) error {
 					}
 
 				case *ast.CallExpr:
-					// continues below
-				default:
-					return true
+					// Reverse position information of call sites.
+					pos := fset.Position(node.Pos())
+					origPos := fmt.Sprintf("%s:%d", goFile, pos.Offset)
+					newFilename := hashWith(lpkg.GarbleActionID, origPos) + ".go"
+
+					// Do "obfuscated.go:1", corresponding to the call site's line.
+					// Most common in stack traces.
+					replaces = append(replaces,
+						newFilename+":1",
+						fmt.Sprintf("%s/%s:%d", lpkg.ImportPath, goFile, pos.Line),
+					)
+
+					// Do "obfuscated.go" as a fallback.
+					// Most useful in build errors in obfuscated code,
+					// since those might land on any line.
+					// Any ":N" line number will end up being useless,
+					// but at least the filename will be correct.
+					replaces = append(replaces,
+						newFilename,
+						fmt.Sprintf("%s/%s", lpkg.ImportPath, goFile),
+					)
```

Comment:
I don't quite understand what you are doing here.

Reply:
a func call at `foo.go:12` gets obfuscated to something like `abcd.go:1`, so we replace it back with the append above. that's what the previous code already did.

what the previous code didn't do is if `abcd.go` appears in some other way, like without a line number suffix, or if a build error happens on a random line without a call site so we end up with `abcd.go:3`.

Reply:
I tried to explain this in the code comment, but it's hard to explain briefly. any suggestions welcome.

Reply:
Ok I get it now